### PR TITLE
maint: fix code style, check on gha

### PR DESCRIPTION
**Pull request by dweindl**
Follow-up to #251 

* Manual changes for compatibility with selected ruff rules
* run pre-commit hooks on GHA instead of flake8
* run tests on python3.12

### CODE REVIEW

File: petab/observables.py [hunk 1]
```diff
@@ -55,7 +55,7 @@ def get_observable_df(
     except KeyError:
         raise KeyError(
             f"Observable table missing mandatory field {OBSERVABLE_ID}."
-        )
+        ) from None
```
Comment (dilpath) [inline petab/observables.py:58]: `as err`, `from err`?

File: petab/conditions.py [hunk 2]
```diff
@@ -47,7 +47,7 @@ def get_condition_df(
     except KeyError:
         raise KeyError(
             f"Condition table missing mandatory field {CONDITION_ID}."
-        )
+        ) from None
```
Comment (dilpath) [inline petab/conditions.py:50]: `as err`, `from err`?

File: setup.py [hunk 3]
```diff
@@ -83,7 +83,7 @@ def absolute_links(txt):
             "pysb",
         ],
         "quality": [
-            "flake8>=3.8.3",
+            "pre-commit",
```
Comment (dilpath) [inline setup.py:86]: Is the `ruff` dependency somehow managed by `pre-commit` and the ruff-pre-commit repo?

https://github.com/astral-sh/ruff-pre-commit/blob/e76e45000a5aa92cd2ec4d3fb1485f2f9fea5097/pyproject.toml#L1-L6

File: tests/test_petab.py [hunk 4]
```diff
@@ -382,17 +384,23 @@ def test_flatten_timepoint_specific_output_overrides():
...
             OBSERVABLE_FORMULA: [
-                f"observableParameter1_{obs1_1_1_1} + observableParameter2_{obs1_1_1_1}",
-                f"observableParameter1_{obs1_2_1_1} + observableParameter2_{obs1_2_1_1}",
-                f"observableParameter1_{obs1_2_2_1} + observableParameter2_{obs1_2_2_1}",
+                f"observableParameter1_{obs1_1_1_1}"
+                f" + observableParameter2_{obs1_1_1_1}",
+                f"observableParameter1_{obs1_2_1_1}"
+                f" + observableParameter2_{obs1_2_1_1}",
+                f"observableParameter1_{obs1_2_2_1}"
+                f" + observableParameter2_{obs1_2_2_1}",
```
Comment (dilpath) [inline tests/test_petab.py:392]: :neutral_face:

**Review by dilpath — APPROVED** [review]
:+1:

File: petab/observables.py (hunk 1 above)
Comment (dweindl) [reply petab/observables.py:58]: I'd prefer `from None`. We only want this simple message here without the pandas error on top.

File: petab/conditions.py (hunk 2 above)
Comment (dweindl) [reply petab/conditions.py:50]: as previous

File: setup.py (hunk 3 above)
Comment (dweindl) [reply setup.py:86]: correct

File: tests/test_petab.py (hunk 4 above)
Comment (dweindl) [reply tests/test_petab.py:392]: we should have thought of something shorter than "observableParameter1_" :sweat_smile: :see_no_evil:

**Pull request merged**